utils/watermark_disk_test.go

```diff
@@ -0,0 +1,87 @@
+package utils
+
+import (
+	"crypto/rand"
+	"os"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+func Test_ApplyWatermarks(t *testing.T) {
+	// Create a temporary file
+	tempFile, err := os.CreateTemp("", "testfile")
+	if err != nil {
+		t.Fatalf("Failed to create temporary file: %v", err)
+	}
+	defer os.Remove(tempFile.Name())
+
+	// Close the file since we'll be reopening it in ApplyWatermarks
+	tempFile.Close()
+
+	t.Run("NegativeTest", func(t *testing.T) {
+		// Create a ~15KB empty file, no room for all watermarks
+		err := os.WriteFile(tempFile.Name(), make([]byte, 1*1024), 0o600)
+		if err != nil {
+			t.Fatalf("Failed to create empty file: %v", err)
+		}
```

Comment:
```suggestion
		// Create a 1KB empty file, no room for all watermarks
		assert.NoError(t, os.Truncate(tempFile.Name(), make(1*1024)))
```

Reply:
This is what you meant, right?

`assert.NoError(t, os.Truncate(tempFile.Name(), 1*1024))`